### PR TITLE
Remove or replace all the @nuxt libraries

### LINKED ISSUE #11272: Remove or replace all the @nuxt libraries
# Description

Remove or replace all the libraries used within the Nuxt ecosystem:
  - `@nuxtjs/axios`, to be replaced with simply `axios`?
  - `@nuxtjs/webpack-profile`, TBD scope
  - ~`cookie-universal-nuxt`, used for 2 way binding between cookies and global $cookies~ https://github.com/rancher/dashboard/issues/11321
  - ~`@nuxt/babel-preset-app`, default Nuxt loaders~ https://github.com/rancher/dashboard/issues/11375
  - `@nuxt/typescript-build`, dead code
  - `@nuxt/types`, dead code
  - `@nuxtjs/eslint-config-typescript`, dead code

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #11272

### Technical notes summary

Strictly libraries related changes, as original plan

- Replaced `@nuxtjs/axios` with `defu`, `axios` and `axios-retry`
- Corrected `http-proxy-middleware` import

Side effects crossed while debugging:
- Pinned ALL the versions and resolutions to current working versions
- Added dependency resolution for `roarr` used in a dependency of `cy2`
- Extended migration script due changes in GH Workflow tree
- Updated checkout@, artifact@ and node@ actions to latest for multiple GHA workflows
- ~Updated Node version in CI to 16 for missing cases after issue on master/rebase https://github.com/rancher/dashboard/pull/11377~ This has been replaced by `.nvmrc` file versioning pinning at this time.
- Extended babel configuration for `@rancher/components` due issue on compile not pointing to the right ES module after removing `@nuxtjs/webpack-profile`

Generic side effects:
- Removed script/style from within the app in Vue index.html
- Abstracted exports from Vue files as they started to fail (bad practice):

Screenshot 2024-07-01 at 16 36 10

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -37,29 +37,31 @@ jobs:
...
-        compression-level: 9
-     - name: Upload e2e build ember
-       uses: actions/upload-artifact@v4
-       with:
-        name: ${{ env.E2E_BUILD_DIST_EMBER_NAME }}
-        path: ${{ env.E2E_BUILD_DIST_EMBER_DIR }}/
-        if-no-files-found: error
-        retention-days: 10
-        compression-level: 9
+      - uses: actions/checkout@v4
```
Comment (cnotv) [inline .github/workflows/test.yaml:40]: Indentation was wrong, bumped checkout versions

**Comment by rak-phillip** [comment]
I'm still seeing some warnings about `window.$nuxt` coming from `shell/initialize/App.vue`

```
window.$nuxt is deprecated. It would be best to stop using globalState all together. For an alternative you can use window.$globalApp.
```

Is this something that we can safely remove at this point?

**Comment by cnotv** [comment]
> I'm still seeing some warnings about `window.$nuxt` coming from `shell/initialize/App.vue`
> 
> ```
> window.$nuxt is deprecated. It would be best to stop using globalState all together. For an alternative you can use window.$globalApp.
> ```
> 
> Is this something that we can safely remove at this point?

It is mentioned to be for some extensions, so not sure if we can. Ideally, older extensions should be blocked.
What do you thing @aalves08 ?

**Comment by aalves08** [comment]
> $nuxt

I've looked into Elemental, Kubewarden, Neuvector, Stackstate, CAPI 🐢 , Kamaji and couldn't find any hits while searching for `$nuxt`. If there are more implications, that I don't know, but there's no explicit usage in extensions code (at least the ones that I searched for) @cnotv

**Comment by cnotv** [comment]
> > $nuxt
> 
> I've looked into Elemental, Kubewarden, Neuvector, Stackstate, CAPI 🐢 , Kamaji and couldn't find any hits while searching for `$nuxt`. If there are more implications, that I don't know, but there's no explicit usage in extensions code (at least the ones that I searched for) @cnotv

@aalves08 @rak-phillip this is about Harvester!
The whole code is the following:

```js
        if (!isHarvester) {
          console.warn('window.$nuxt is deprecated. It would be best to stop using globalState all together. For an alternative you can use window.$globalApp.'); // eslint-disable-line no-console
        }
```

@torchiaf do we know if this is still needed?
We have this issue where we introduced it to avoid the warning, is it still relevant?
https://github.com/rancher/dashboard/issues/10681

**Comment by torchiaf** [comment]
> @torchiaf do we know if this is still needed? We have this issue where we introduced it to avoid the warning, is it still relevant? #10681

Let's remove it.

**Comment by cnotv** [comment]
> I'm still seeing some warnings about `window.$nuxt` coming from `shell/initialize/App.vue`
> 
> ```
> window.$nuxt is deprecated. It would be best to stop using globalState all together. For an alternative you can use window.$globalApp.
> ```
> 
> Is this something that we can safely remove at this point?

We finally agreed about removing it and is out now.

**Review by rak-phillip — DISMISSED** [review]
Functionally, this looks good to me. 

It looks like there's a minor merge conflict that needs to be resolved.

File: package.json [hunk 2]
```diff
@@ -136,6 +136,7 @@
   },
   "devDependencies": {
     "@babel/plugin-proposal-optional-chaining": "7.14.5",
+    "@babel/plugin-proposal-private-methods": "^7.18.6",
```
Comment (rak-phillip) [inline package.json:139]: Shall we remove the `^` to ensure that we match the version exactly, like we do with the rest of our dependencies?

Comment (cnotv) [reply package.json:139]: Absolutely, thanks for the catch!

**Review by rak-phillip — APPROVED** [review]

**Comment by cnotv** [comment]
> > I'm still seeing some warnings about `window.$nuxt` coming from `shell/initialize/App.vue`
> > ```
> > window.$nuxt is deprecated. It would be best to stop using globalState all together. For an alternative you can use window.$globalApp.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Is this something that we can safely remove at this point?
> 
> ~We finally agreed about removing it and is out now.~ We are going to test in a `rancher/dashboard` branch to ensure Harvester does not break: https://github.com/rancher/dashboard/tree/nuxt-removal-dev

It looks like we are clashing with working time, so since we can test it afterward on the branch, we'll just merge it. If any issue we'll restore this part of the code or solve in Harvester I guess.

**Comment by torchiaf** [comment]
I've tested the navigation from/to Harvester clusters and navigating Harvester pages.
I get this error message in the console + navigation inside Harvester is broken:

image

How to replicate:

- Click on Virtualization Management
- Click on a Harvester cluster
- Go to Virtual Machine page

**Comment by cnotv** [comment]
> @cnotv @rak-phillip I've tested the navigation from/to Harvester clusters and navigating Harvester pages. I get this error message in the console + navigation inside Harvester is broken:
> 
> How to replicate:
> 
> * Click on Virtualization Management
> * Click on a Harvester cluster
> * Go to Virtual Machine page

Ok I'll make a PR to restore that part and add a comment about it.

**Comment by cnotv** [comment]
I can confirm that the issue is related to this last code removed.

Screenshot 2024-08-09 at 14 13 12

**Pull request merged**